engine/execution/rpc/engine.go

```diff
@@ -229,6 +241,22 @@ func (h *handler) GetEventsForBlockIDs(_ context.Context,
 		return nil, err
 	}
 
+	if len(blockIDs) > h.maxBlockRange {
+		return nil, status.Errorf(codes.InvalidArgument, "too many block IDs requested")
+	}
+
+	// make sure all blocks exist first to fail fast and avoid unused lookups
+	// must verify block exists first, since ByBlockID* calls on sets like events or commits will
+	// return an empty slice if block does not exist
+	for _, blockID := range flowBlockIDs {
+		if _, err = h.headers.ByBlockID(blockID); err != nil {
```

Comment:
I don't think it's necessary, because `h.commits.ByBlockID(bID)` is also a lightweight check to ensure a block has been executed, if a block has been executed, it must exist. If the block exists but has not been executed, we also want to return not found, which means this check isn't necessary.